### PR TITLE
bpo-37775: update docs of compileall

### PR DESCRIPTION
**Pull request by shihai1991**
https://bugs.python.org/issue37775

### CODE REVIEW

**Comment by shihai1991** [comment]
cc @vstinner @benjaminp

File: Doc/library/compileall.rst [hunk 1]
```diff
@@ -232,7 +232,7 @@ Public functions
    .. versionchanged:: 3.7
       The *invalidation_mode* parameter was added.
 
-.. function:: compile_path(skip_curdir=True, maxlevels=0, force=False, quiet=0, legacy=False, optimize=-1, invalidation_mode=py_compile.PycInvalidationMode.TIMESTAMP)
+.. function:: compile_path(skip_curdir=True, maxlevels=0, force=False, quiet=0, legacy=False, optimize=-1, invalidation_mode=None)
```
Comment (CuriousLearner) [inline Doc/library/compileall.rst:241]: FWIW, these should probably also contain a `versionchanged` entry as done in https://github.com/python/cpython/commit/a6b3ec5b6d4f6387820fccc570eea08b9615620d#diff-ff4f9cbddd1c0a6371e3cc85fbea0785R80

This should be done for each of the methods whose signature is now changed, mentioning about the change and how default value is determined.

**Review by CuriousLearner — CHANGES_REQUESTED** [review]
I have a suggestion about these changes. Please find my comments inline.

**Comment by CuriousLearner** [comment]
This pull request also does not require a NEWS entry and a core-developer would be able to add Skip News label that should take care of the current failing status for `bedevere/news` bot.

File: Doc/library/compileall.rst (hunk 1 above)
Comment (shihai1991) [reply Doc/library/compileall.rst:241]: thanks for review, Done

File: Doc/library/compileall.rst [hunk 2]
```diff
@@ -185,10 +185,13 @@ Public functions
...
       The *invalidation_mode* parameter was added.
 
+   .. versionchanged:: 3.7.2
+      The *invalidation_mode* parameter's default value is updated to None.
+
    .. versionchanged:: 3.8
       Setting *workers* to 0 now chooses the optimal number of cores.
 
-.. function:: compile_file(fullname, ddir=None, force=False, rx=None, quiet=0, legacy=False, optimize=-1, invalidation_mode=py_compile.PycInvalidationMode.TIMESTAMP)
+.. function:: compile_file(fullname, ddir=None, force=False, rx=None, quiet=0, legacy=False, optimize=-1, invalidation_mode=None)
```
Comment (CuriousLearner) [inline Doc/library/compileall.rst:194]: It seems the `versionchanged` tag was to be added for this method as well.

Comment (shihai1991) [reply Doc/library/compileall.rst:194]: ops, i forgot this line. Thanks, Sanyam :)

**Review by CuriousLearner — APPROVED** [review]
LGTM 🌮

**Comment by vstinner** [comment]
@shihai1991: There is a conflict in the 3.7 backport. Can you please try to create a PR for Python 3.7 using "git cherry-pick -x" (and fix conflicts)?

**Pull request merged**